eventdisplay/FairEventManagerEditor.cxx

```diff
@@ -111,77 +121,114 @@ void FairEventManagerEditor::Init()
     f2->AddFrame(fEventTime);
     title1->AddFrame(f2);
 
-    fVizPri = new TGCheckButton(title1, "Primary Only");
-    fVizPri->Connect("Toggled(Bool_t)", "FairEventManagerEditor", this, "DoVizPri()");
-    title1->AddFrame(fVizPri);
-
-    TGHorizontalFrame* f1 = new TGHorizontalFrame(title1);
-    TGLabel* L1 = new TGLabel(f1, "Select PDG :");
-    f1->AddFrame(L1, new TGLayoutHints(kLHintsLeft | kLHintsCenterY, 1, 2, 1, 1));
-    fCurrentPDG = new TGNumberEntry(
-        f1, 0., 12, -1, TGNumberFormat::kNESInteger, TGNumberFormat::kNEAAnyNumber, TGNumberFormat::kNELNoLimits, 0, 1);
-    f1->AddFrame(fCurrentPDG, new TGLayoutHints(kLHintsLeft, 1, 1, 1, 1));
-    fCurrentPDG->Connect("ValueSet(Long_t)", "FairEventManagerEditor", this, "SelectPDG()");
-    title1->AddFrame(f1);
-
-    fMinEnergy = new TEveGValuator(title1, "Min Energy:", 90, 0);
-    fMinEnergy->SetNELength(5);
-    fMinEnergy->SetLabelWidth(80);
-    fMinEnergy->Build();
-    fMinEnergy->SetLimits(0, MAXE, 2501, TGNumberFormat::kNESRealOne);
-    fMinEnergy->SetToolTip("Minimum energy of displayed track.");
-    fMinEnergy->Connect("ValueSet(Double_t)", "FairEventManagerEditor", this, "MinEnergy()");
-    title1->AddFrame(fMinEnergy, new TGLayoutHints(kLHintsTop, 1, 1, 1, 0));
-
-    fMaxEnergy = new TEveGValuator(title1, "Max Energy:", 90, 0);
-    fMaxEnergy->SetNELength(5);
-    fMaxEnergy->SetLabelWidth(80);
-    fMaxEnergy->Build();
-    fMaxEnergy->SetLimits(0, MAXE, 2501, TGNumberFormat::kNESRealOne);
-    fMaxEnergy->SetToolTip("Maximum energy of displayed track.");
-    fMaxEnergy->SetValue(MAXE);
-    fMaxEnergy->Connect("ValueSet(Double_t)", "FairEventManagerEditor", this, "MaxEnergy()");
-    title1->AddFrame(fMaxEnergy, new TGLayoutHints(kLHintsTop, 1, 1, 1, 0));
-
     TGTextButton* fUpdate = new TGTextButton(title1, "Update");
     title1->AddFrame(fUpdate, new TGLayoutHints(kLHintsRight | kLHintsExpandX, 5, 5, 1, 1));
     fUpdate->Connect("Clicked()", "FairEventManagerEditor", this, "SelectEvent()");
 
     fInfoFrame->AddFrame(title1, new TGLayoutHints(kLHintsTop, 0, 0, 2, 0));
 
     fManager->SetMaxEnergy(MAXE);
-}
 
-void FairEventManagerEditor::MaxEnergy() { fManager->SetMaxEnergy(fMaxEnergy->GetValue()); }
+    //=============== graphics =============================
+    TGVerticalFrame* scene_conf = CreateEditorTabSubFrame("Graphics");
+    TGHorizontalFrame* transparency_frame = new TGHorizontalFrame(scene_conf);
 
-void FairEventManagerEditor::MinEnergy() { fManager->SetMinEnergy(fMinEnergy->GetValue()); }
+    auto transparency(
+        std::unique_ptr<FairEveTransparencyControl>(new FairEveTransparencyControl(scene_conf, "Global transparency")));
```

Comment:
```suggestion
    std::unique_ptr<FairEveTransparencyControl> transparency(new FairEveTransparencyControl(scene_conf, "Global transparency"));
```

or

```suggestion
    auto transparency(fair::make_unique<FairEveTransparencyControl>(scene_conf, "Global transparency"));
```

For the latter please cherry-pick https://github.com/dennisklein/FairRoot/commit/d7bfe17446f87483d287857c761af3861cc8a58a and add `#include "FairSmartPointer.h"`. This is just a backport of a helper that was added to the standard library in C++14 ([std::make_unique](https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique)). See also [C.150: Use make_unique() to construct objects owned by unique_ptrs](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c150-use-make_unique-to-construct-objects-owned-by-unique_ptrs).

I am sorry for being pedantic about this, but this is framework code and I want to avoid people copy&pasting bad examples (we already have plenty of anti-patterns that get copied around like crazy). We should either use `unique_ptr`, `auto` etc the right way or not at all.